Automattic/wp-calypso
# Gutenberg: refactor Calypsoify code for Atomic sites

#### Changes proposed in this Pull Request

* Bump version check to `5.0` - since leaving RC there was confusing.
* Refactoring WordPress version selector to use `getSiteOptions` (props @kwight)
* Adding the Classic plugin check for Calypsoify Atomic switching logic.

#### Testing instructions

In order to make testing easier, let's wait until all Atomic sites are updated to `5.0` and things are stable before landing this refactor.

- `npm run test-client client/state/selectors/test/get-wordpress-version.js`
- `ENABLE_FEATURES=calypsoify/gutenberg npm start`
- Prepare Two Simple Sites with a Business plan
- For the first make sure `gutenberg` is the default editor
- For the second make sure `classic` is the default editor
- Easiest way to do this is by visiting /post and clicking on the opt-in nudge on the sidebar
<img width="287" alt="screen shot 2018-12-05 at 2 52 49 pm" src="https://user-images.githubusercontent.com/1270189/49549452-a6652300-f89d-11e8-9dde-75602e32ca5b.png">

- Upload a plugin or theme on each site
- Starting from Calypso click on any post edit link
We should see:
<img width="1143" alt="screen shot 2018-12-05 at 2 26 59 pm" src="https://user-images.githubusercontent.com/1270189/49548420-8849f380-f89a-11e8-909a-35cdc19fdf3d.png">

- Note that we **don't** have a switch back to classic option in the more dropdown.

- No regressions for Gutenberg Calypsoify flows for Simple Sites.

- On test site that has opted in to Gutenberg install the Classic plugin.

- Verify that Calypsoify no longer redirects to wp-admin for this site. They should see Calypso Classic editor instead.
